### ISSUE TITLE
min select policy in HPA does not trigger scale down behavior 

### ISSUE THREAD

**Issue by varshachandanb** [kubernetes/kubernetes#97468]
**What happened**:
```
spec:
  maxReplicas: 20
  minReplicas: 2
  behavior:
    scaleDown:
      stabilizationWindowSeconds: 300
      policies:
      - type: Percent
        value: 10
        periodSeconds: 60
      - type: Pods
        value: 4
        periodSeconds: 60
      selectPolicy: Min   
    scaleUp:
      stabilizationWindowSeconds: 0
      policies:
      - type: Percent
        value: 10
        periodSeconds: 60
      - type: Pods
        value: 4
        periodSeconds: 60
      selectPolicy: Max
```
HPA does not trigger scale down when `selectPolicy: Min` in behavior. 
Noticed this message on describe HPA: 
"ScalingLimited  True    ScaleDownLimit  the desired replica count is decreasing faster than the maximum scale rate"

**What you expected to happen**:
When the Current/Target metric ratio is lesser than 1, expected scale down to be triggered by reducing 10% percent of pods i.e current pods 20 > 18 (scale down 10% of 20 with min policy) 

**How to reproduce it (as minimally and precisely as possible)**:
Apply the above spec in HPA configuration and maintain Current/Target metric ratio < 1 and notice that scale down does not get triggered at all. 

**Anything else we need to know?**:
Scale down scenario was tested for SQS queue length metric. 

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-18T14:36:53Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: AWS

**Comment by troy0820** [comment]
Related incident https://github.com/kubernetes/kubernetes/issues/97474

**Comment by arjunrn** [comment]
Fixed with #95647 and cherrypicked into v1.18.11

**Comment by varshachandanb** [comment]
The above fix seems to be working. Thanks
